### PR TITLE
Quentinb/feat/entities

### PR DESCRIPTION
**Pull request by QuentinBouygues**
We added the entities to the back end. 
We did not add the cascades, for we will define them as we create the Business Logic.

### CODE REVIEW

**Comment by QuentinBouygues** [comment]
We did not add properties to user-team and team-board associations

I do not remember, should we commit the package-lock file ?

File: src/entities/attachment.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    type: string
+
+    @Column('text')
+    @IsFQDN()
+    URL: string
+
+// ------------------------------------
+//            EXTERNAL LINKS
+// ------------------------------------
+    @ManyToOne(type => Card, card => card.attachments) // Many Attachment to One Card
```
Comment (YvesLongchamp) [inline src/entities/attachment.ts]: Seems obvious, like said Alexis. You don't need to say what it is doing ;)

File: src/entities/board.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            referencedColumnName: 'id'
+        },
+        inverseJoinColumn: {
+            name: 'user',
+            referencedColumnName: 'id'
+        },
+    })
+    users: User[]
+
+    @OneToMany(type => List, list => list.board) // One Board to many Lists
```
Comment (YvesLongchamp) [inline src/entities/board.ts]: Same that above, seems obvious

**Review by YvesLongchamp — CHANGES_REQUESTED** [review]
Most of the comments, for the one to many are repetitive. Change that, and I think this will be fine

File: package.json [hunk 3]
```diff
@@ -43,6 +43,7 @@
   "dependencies": {
     "bcrypt-nodejs": "^0.0.3",
     "body-parser": "^1.15.2",
+    "class-validator": "^0.7.2",
```
Comment (Porquepix) [inline package.json:46]: No `@types/class-validator`, is it normal ?

File: src/entities/board.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    @Column('text')
+    title: string
+
+    @Column('boolean')
+    isPrivate: boolean
+
+// ------------------------------------
+//            EXTERNAL LINKS
+// ------------------------------------
+    @OneToOne(type => Team)
```
Comment (Porquepix) [inline src/entities/board.ts:24]: No reverse relation ?

File: src/entities/card.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import { Entity, Column, PrimaryGeneratedColumn, ManyToMany, JoinTable, ManyToOne, OneToMany } from 'typeorm'
+import { Tag } from './tag'
+import { List } from './list'
+import { TaskList } from './taskList'
+import { Comment } from './comment'
+import { Attachment } from './attachment'
+
+@Entity()
+export class Card {
```
Comment (Porquepix) [inline src/entities/card.ts:9]: Where is the title ?

Comment (Porquepix) [reply src/entities/card.ts:9]: And the checklists ?

File: src/entities/comment.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    @PrimaryGeneratedColumn()
+    id: number
+
+    @Column({
+        type: 'text',
+        length: 25
+    })
+    title: string
+
+    @Column({
```
Comment (Porquepix) [inline src/entities/comment.ts:19]: Why a rank and not a created date ?

File: src/entities/card.ts (hunk 5 above)
Comment (Porquepix) [reply src/entities/card.ts:9]: Card may have a rank too ?

File: src/entities/tag.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        type: 'text',
+        length: 25
+    })
+    label: string
+
+    @Column({
+        type: 'text',
+        length: 25
+    })
+    color: string
```
Comment (Porquepix) [inline src/entities/tag.ts]: Is it possible to have an Enum ?

File: src/entities/task.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import { Entity, Column, PrimaryGeneratedColumn, ManyToOne } from 'typeorm'
+import { TaskList } from './taskList'
+
+@Entity()
+export class Task {
```
Comment (Porquepix) [inline src/entities/task.ts:5]: I think there is a confusion between task and tasklist ?
Task must have a rank and a isDone attribute. TaskList doesn't need isDone.

File: src/entities/taskList.ts [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+        type: 'text',
+        length: 25
+    })
+    title: string
+
+    @Column({
+        type: 'text',
+        length: 500
+    })
+    description: string
```
Comment (Porquepix) [inline src/entities/taskList.ts]: We don't care about a description here

File: src/entities/task.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+        type: 'text',
+        length: 25
+    })
+    title: string
+
+    @Column({
+        type: 'text',
+        length: 500
+    })
+    description: string
```
Comment (Porquepix) [inline src/entities/task.ts]: We don't care about a description here

File: src/entities/card.ts (hunk 5 above)
Comment (QuentinBouygues) [reply src/entities/card.ts:9]: CheckList is declared as TaskList

File: package.json (hunk 3 above)
Comment (QuentinBouygues) [reply package.json:46]: Did not find it, can you try ?

File: src/entities/tag.ts (hunk 7 above)
Comment (QuentinBouygues) [reply src/entities/tag.ts]: Je stocke direct les codes Hexa ?

Comment (QuentinBouygues) [reply src/entities/tag.ts]: Done, added the color names associated to their hexa codes. I also added enums for roles

**Review by YvesLongchamp — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (Porquepix) [reply package.json:46]: Ok, class-validator is already a Typescript lib :)

**Review by Porquepix — APPROVED** [review]

**Pull request merged**